### PR TITLE
feat: add critical tag to S3 modules

### PR DESCRIPTION
**Pull request by gcharest**
# Summary | Résumé

Attempt to add the `Critical:false` tag to all S3 buckets using the S3 modules.

### CODE REVIEW

**Comment by gcharest** [comment]
Partially addresses https://github.com/cds-snc/security-risk-register/issues/47

**Review by patheard — COMMENTED** [review]
I like this idea to prevent accidental deletion.  Was the plan to add the value of these two variables to the bucket tags here?
https://github.com/cds-snc/terraform-modules/blob/7c7738acbbb70d4297c91e1799b02df3e374fdf2/S3/locals.tf#L2-L5

**Comment by gcharest** [comment]
@patheard Yes, the goal would be to be able to enable this tag and create the proper SCPs pulling resources type + tag to apply them. Not there yet though, just at the flag stage atm!

File: S3/input.tf [hunk 1]
```diff
@@ -12,6 +12,18 @@ variable "billing_tag_value" {
...
 
+variable "critical_tag_key" {
+  description = "(Optional) The value of the critical tag. If set to true, protection SCP rules will be applied to the bucket."
+  type        = string
+  default     = "Critical"
+}
+
+variable "critical_tag_value" {
+  description = "(Required, default false) The value of the critical tag. If set to true, protection SCP rules will be applied to the bucket."
+  type        = string
```
Comment (CalvinRodo) [inline S3/input.tf:23]: Change this to `boolean`

File: S3/locals.tf [hunk 2]
```diff
@@ -1,6 +1,7 @@
 locals {
   common_tags = {
-    (var.billing_tag_key) = var.billing_tag_value
-    Terraform             = "true"
+    (var.billing_tag_key)  = var.billing_tag_value
+    Terraform              = "true"
+    (var.critical_tag_key) = var.critical_tag_value
```
Comment (CalvinRodo) [inline S3/locals.tf:5]: (var.critical_tag_key) = var.critical_tag_value ? "true": "false"

**Review by CalvinRodo — CHANGES_REQUESTED** [review]
See changes about input.

File: S3/input.tf (hunk 1 above)
Comment (gcharest) [reply S3/input.tf]: done

File: S3/locals.tf (hunk 2 above)
Comment (gcharest) [reply S3/locals.tf]: updated as discussed

**Review by CalvinRodo — APPROVED** [review]
:taco::cat:

**Comment by gcharest** [comment]
@CalvinRodo I borked the rebased to sign the very first commit so had to do a forced push, erasing the history of the previous commits.

**Pull request merged**